File: loja/pedidos/src/main/java/br/com/bota/ws/low/ClienteClient.java
package br.com.bota.ws.low;

import br.com.bota.dto.ClienteDTO;
import io.micronaut.core.type.Argument;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.client.RxHttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.http.uri.UriBuilder;
import io.reactivex.Flowable;
import io.reactivex.Maybe;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.UUID;

@Singleton
public class ClienteClient {
    @Inject
    @Client("${api.url.cliente}")
    private RxHttpClient httpClient;

   public Maybe<ClienteDTO> buscarClientePorId(UUID clienteId) {
        HttpRequest<?> req = HttpRequest.GET(UriBuilder.of(clienteId.toString()).build());
        Flowable flowable = httpClient.retrieve(req, Argument.of(ClienteDTO.class));
        return flowable.firstElement();
    }
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/ws/lowapi/ClienteClient.kt
package br.com.bota.ws.lowapi

import br.com.bota.lojalib.dto.ClienteDTO
import io.micronaut.core.type.Argument
import io.micronaut.http.HttpRequest
import io.micronaut.http.client.HttpClient
import io.micronaut.http.client.annotation.Client
import io.micronaut.http.uri.UriBuilder
import io.reactivex.Flowable
import io.reactivex.Maybe
import jakarta.inject.Inject
import jakarta.inject.Singleton
import java.util.*

@Singleton
open class ClienteClient @Inject constructor(@param:Client("\${api.url.cliente}")  val httpClient: HttpClient) {
    fun buscarClientePorId(clienteId: UUID) : Maybe<ClienteDTO> {
        val req:HttpRequest<ClienteDTO> = HttpRequest.GET(UriBuilder.of(clienteId.toString()).build())
        val flowable = Flowable.fromPublisher(httpClient.retrieve(req, Argument.of(ClienteDTO::class.java)))
        return flowable.firstElement()
    }
}